### ISSUE TITLE
Style/SuperArguments incompatible with define_method

### ISSUE THREAD

**Issue by HoneyryderChuck** [rubocop/rubocop#12957]
## Expected behavior

The recently inntroduced `Style/SuperArguments` should "just work". I assumed that, by just autocorrecting [the errors from this test suite](https://gitlab.com/os85/rodauth-oauth/-/jobs/6944724946), the errors would go away. 

## Actual behavior

Replacing `super` calls broke the code. `rodauth` defines auth methods using `define_method`, which does not support implicit `super` calls:

```
RodauthOauthDynamicClientRegistrationTest#test_oauth_dynamic_client_response_types:
RuntimeError: implicit argument passing of super from method defined by define_method() is not supported. Specify all arguments explicitly.
```

## Steps to reproduce the problem

clone the rodauth-oauth repository, enable the cop, run `rubocop -a lib test` and run `bundle exec rake test` to find the offending tests.

## RuboCop version

`rubocop-1.64.1`

**Comment by Earlopain** [comment]
There's no way that RuboCop can catch this case, it must be marked as unsafe. Something like this is already not an offense:

```rb
def generate_method(a)
  define_method(:foo) { super(a) }
end
```

But something like this is just not possible, I assume something like that happens in your repo (I must admit that the code flow even with stacktrace was not obvious to me):
```rb
def generate_method(a)
  context_switch { super(a) }
end

def context_switch(&)
  define_method(:foo, &)
end
```

**Comment by HoneyryderChuck** [comment]
@Earlopain I believe you're looking for [this](https://github.com/jeremyevans/rodauth/blob/master/lib/rodauth.rb#L80).

**Comment by HoneyryderChuck** [comment]
Thx everyone 🙏
